api/krusty/localizetests/localizerunner.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package localizetests
+
+import (
+	"sigs.k8s.io/kustomize/api/internal/localizer"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+// LocalizeRunner runs `kustomize localize`. Like a CLI subprocess executing the command,
+// it only needs the command arguments and flags and to be injected with a file system to run.
+type LocalizeRunner struct {
+	Scope string
+}
+
+// Run `localize`s referenced files in kustomization target to destination newDir on fSys.
+func (lr *LocalizeRunner) Run(fSys filesys.FileSystem, target string, newDir string) error {
```

Comment:
Giving this a little more thought, and it seems like the primary purpose of `LocalizeRunner` is to store the `scope`. Is there a clear benefit of this? i.e.

```
(&LocalizeRunner{Scope: "scope"}).Run(fs, target, dst)
```

vs just using the localizer directly:

```
localizer.Run(target, "scope", dst, fs)
```

The latter seems simpler.

Reply:
I removed `LocalizeRunner` and changed `localizetests` to `localizer`.

I originally meant for `LocalizeRunner` to resemble `kustomizer`: https://github.com/kubernetes-sigs/kustomize/blob/master/api/krusty/kustomizer.go#L32, which stores all command flags so that `Run` only needs to be passed the command argument. I understand if you think this structure doesn't make sense for `localize` because `localize` only has 1 flag now. 

I can't remove the `Run` wrapper code from `krusty` completely because `localizer` lives in `api/internal` and so the command code in `kustomize/commands` doesn't have access to it.

I changed the package name `localizetests` because, after consideration, the main code here is the wrapper `Run`, not the tests. `localizetests.Run` seems counter-intuitive to me.